### PR TITLE
Add helper type TextMapCarrier; Add "error" tag

### PR DESCRIPTION
**Pull request by yurishkuro**
@bensigelman I think this is not controversial

### CODE REVIEW

File: propagation.go [hunk 1]
```diff
@@ -112,8 +112,27 @@ type TextMapReader interface {
 	ForeachKey(handler func(key, val string) error) error
 }
 
+// TextMapCarrier allows the use of regular map[string]string
+// as both TextMapWriter and TextMapReader.
```
Comment (bensigelman) [inline propagation.go]: technically this is in violation of the multimap spec, though the multimap spec is about to change... what to do??

**Comment by bensigelman** [comment]
@yurishkuro this LGTM in the broader sense, though at the moment it does not fulfill the contract (per my comment). Thoughts?

File: propagation.go (hunk 1 above)
Comment (yurishkuro) [inline propagation.go]: @bensigelman  I don't think we have multi-map *requirement* in the current API. The Writer doc says

```
	// Set a key:value pair to the carrier. Multiple calls to Set() for the
	// same key leads to undefined behavior.
```
so `map[string]string` satisfies that by overriding the key.

And the reader API says multiple keys *may* be returned, but map wouldn't do it.
```
	// NOTE: A single `key` may appear in multiple calls to `handler` for a
	// single `ForeachKey` invocation.
```

**Comment by bensigelman** [comment]
:ship:

**Pull request merged**